hax/hax/util.py

```diff
@@ -639,33 +639,119 @@ def get_node_fid(self, node: str) -> Optional[Fid]:
         return None
 
     @repeat_if_fails()
-    def get_node_ctrl_fid(self, node: str) -> Optional[Fid]:
+    def get_node_ctrl_fids(self, node: str) -> Optional[List[Fid]]:
         """
-        Returns the fid of the controller for the given node.
-
         Parameters:
             node : hostname of the node.
         """
         # Example,
-        # {
+        # [
         #    "key": "m0conf/sites/0x5300000000000001:0x1/
         #            racks/0x6100000000000001:0x2/encls/
         #            0x6500000000000001:0x4/ctrls/0x6300000000000001:0x5",
-        # },
+        #    "key": "m0conf/sites/0x5300000000000001:0x1/
+        #            racks/0x6100000000000001:0x2/encls/
+        #            0x6500000000000001:0x4/ctrls/0x6300000000000001:0x5",
+        # ]
         encl_fid = self.get_node_encl_fid(node)
         if not encl_fid:
             return None
         ctrl_items = self.kv.kv_get('m0conf/sites', recurse=True)
         regex = re.compile(
             f'^m0conf\\/.*\\/racks\\/.*\\/encls\\/{encl_fid}\\/ctrls\\/'
             '([^/]+)$')
+        list_fids: List[Fid] = []
         for ctrl in ctrl_items:
             match_result = re.match(regex, ctrl['Key'])
             if not match_result:
                 continue
             ctrl_fid: str = match_result.group(1)
+            list_fids.append(Fid.parse(ctrl_fid))
+        return list_fids
+
+    @repeat_if_fails()
+    def get_ioservice_ctrl_fid(self, ioservice_fid: Fid,) -> Optional[Fid]:
```

Comment:
@mssawant  comment: 
add TODO:
Instead of comparing devices, add a mapping from controller <-> I/O service

Reply:
Done